### PR TITLE
Desktop: Fix to update tag list when note is deleted.

### PR DESCRIPTION
**Pull request by mic704b**
The tags shown under the edit bar are not updated when the note is deleted.

## Steps to reproduce
1. Create 2 notes (`A`, `B`), and tag each of them with a different tag (`1`, `2`)
2. Select note `A` in the note list
3. Delete note `A`

Expected result:  note`B` is shown with tag `2`
Observed result: note `B` is shown with tag `1`

## Comments
The small fix is made in the `reduxSharedMiddleware`.  Integration tests added, but only to check the fix.

### CODE REVIEW

File: ReactNativeClient/lib/models/Setting.js [hunk 1]
```diff
@@ -652,8 +652,8 @@ class Setting extends BaseModel {
 
 				this.cache_.push(c);
 			}
-
 			this.dispatchUpdateAll();
+
 		});
 	}
```
Comment (laurent22) [inline ReactNativeClient/lib/models/Setting.js]: I think you could remove this file from the commit as it's mostly white space changes.

File: ReactNativeClient/lib/reducer.js [hunk 2]
```diff
@@ -446,15 +446,14 @@ const reducer = (state = defaultState, action) => {
...
 			newState.notes = action.notes;
 			newState.notesSource = action.notesSource;
 			break;
 
-			// Insert the note into the note list if it's new, or
-			// update it within the note array if it already exists.
+		// Insert the note into the note list if it's new, or
+		// update it within the note array if it already exists.
 		case 'NOTE_UPDATE_ONE':
 			{
```
Comment (laurent22) [inline ReactNativeClient/lib/reducer.js]: Also please remove this file.

**Comment by laurent22** [comment]
Looks good, other than the minor comments I've left. Thanks for the tests you've been adding lately @mic704b, that's extremely useful. It will improve the quality of the apps and, since you've setup the framework, others will be able to add new tests easily.

File: ReactNativeClient/lib/models/Setting.js (hunk 1 above)
Comment (mic704b) [reply ReactNativeClient/lib/models/Setting.js]: Done

File: ReactNativeClient/lib/reducer.js (hunk 2 above)
Comment (mic704b) [reply ReactNativeClient/lib/reducer.js]: Done

**Comment by mic704b** [comment]
Thanks @laurent22!

**Comment by laurent22** [comment]
Perfect, thanks @mic704b!

**Pull request merged**